File: 2640-find-the-score-of-all-prefixes-of-an-array/2640-find-the-score-of-all-prefixes-of-an-array.cpp
class Solution {
public:
    vector<long long> findPrefixScore(vector<int>& nums) {
      vector<long long> pre;
      long long m = 0, s = 0;
      for(int x : nums) {
          m = max(m, (long long)x);
          s += x + m;
          pre.push_back(s);
      }
      return pre;
  }
};

File: 2640-find-the-score-of-all-prefixes-of-an-array/2640-find-the-score-of-all-prefixes-of-an-array.java
class Solution {
    public long[] findPrefixScore(int[] nums) {
        int n = nums.length, maximum = nums[0];
        long[] result = new long[n];
        result[0] = 2 * nums[0];
        for (int i = 1; i < n; i++) {
            maximum = Math.max(maximum, nums[i]);
            result[i] = result[i - 1] + nums[i] + maximum;
        }
        return result;
    }
}